### ISSUE TITLE
resolveType function is not executed 

### ISSUE THREAD

**Issue by eakl** [apollographql/apollo-server#4767]
### Versions
- Apollo-Server: 2.18.2

### Expected behavior
The `__resolveType` function must resolve the `Answer` interface to the proper type (`YearAnswer` or `MonthAnswer` )

### Actual Behavior
The graphQL query throw an error saying the `Answer` type should provide a `resolveType` function.
I do have a `resolveType` function so I believe the function is not executed.

### Issue Reproduction

https://codesandbox.io/s/apollo-server-resolvetype-3x5fo

```graphql
query Questions {
  getQuestions {
    id
    title
    hasAnswers {
      id
    }
  }
}
```

**Comment by eakl** [comment]
### Error 1
On my local machine, `nodemon` provides the following warning:

```bash
Type "Answer" is missing a "__resolveType" resolver.
```

### Error 2
The GraphQL query returns this error:

```json
"message": "Abstract type Answer must resolve to an Object type at runtime for field Question.hasAnswers with value [{ id: 1, year: 2020 }], received \"null\". Either the Answer type should provide a \"resolveType\" function or each possible type should provide an \"isTypeOf\" function."
```

#### Error 3
`console.log('answer:', answer)` inside the `__resolveType` function is not even fired

**Comment by eakl** [comment]
The issue comes from that fact that inside the `resolveType` function, I am trying to access the `parent` argument as an object (like the example provided in the [documentation](https://www.apollographql.com/docs/apollo-server/schema/unions-interfaces/)), while in my example, it is an array

1. Even though my schema expects an array for the `hasAnswer` field, why is the `parent` argument of the `resolveType` function an array. The function is executed for each child so it always returns an array of 1 element, forcing the user to access the parent value with `parent[0]`. Shouldn't the `parent` argument directly be the child?

2. While accessing the `parent` object with `parent[0]`, the graphql query returns `null` for all `hasAnswer` children (I don't see anything wrong in my resolver map for `hasAnswer`)

```graphql
{
  "data": {
    "getQuestions": [
      {
        "id": "1",
        "title": "What is the current year",
        "hasAnswers": [
          {
            "id": null
          }
        ]
      },
      {
        "id": "2",
        "title": "What is the current month",
        "hasAnswers": [
          {
            "id": null
          }
        ]
      }
    ]
  }
}
```

**Comment by eakl** [comment]
In the Question resolver, I didn't join the hasAnswers array with the answers db correctly.

filter() and map() both return an array. I ended up with an array of array that led to the parent argument in the resolveType function being an array and the graphql query returning null for the joined fields.

**I replaced all filter() functions by find()** and everything works as expected.

I'm closing the issue.
